repository.go

```diff
@@ -897,15 +897,29 @@ func (r *Repository) ResolveRevision(rev plumbing.Revision) (*plumbing.Hash, err
 	for _, item := range items {
 		switch item.(type) {
 		case revision.Ref:
-			ref, err := storer.ResolveReference(r.Storer, plumbing.ReferenceName(item.(revision.Ref)))
-
-			if err != nil {
-				return &plumbing.ZeroHash, err
+			revisionRef := item.(revision.Ref)
+			var ref *plumbing.Reference
+
+			for _, rule := range []string{
+				"%s",
+				"refs/%s",
+				"refs/tags/%s",
+				"refs/heads/%s",
+				"refs/remotes/%s",
+				"refs/remotes/%s/HEAD",
```

Comment:
This part exists already in plumbing, I was wondering if it would be possible to move all this code related to revision in plumbing ?

Reply:
Maybe we can make this public:
 https://github.com/src-d/go-git/blob/902347aef9a7cd618c48e3baa1121b72e2beab68/plumbing/reference.go#L18-L27

Reply:
Don't you think we could replace those litteral strings with constants defined just right above to avoid duplication ?